src/curses.rs

```diff
@@ -235,6 +245,10 @@ impl Term {
             clr_eol: get_str_cap(&db, "ce"),
             clr_eos: get_str_cap(&db, "cd"),
 
+            prompt_start: if support_osc133 { Some(CString::new("\x1b]133;A\x1b\x5c").unwrap()) } else { None },
+            prompt_end: if support_osc133 { Some(CString::new("\x1b]133;B\x1b\x5c").unwrap()) } else { None },
+            command_start: if support_osc133 { Some(CString::new("\x1b]133;C\x1b\x5c").unwrap()) } else { None },
```

Comment:
no need to open a new pull request, you can update the old one, or at least cross-reference https://github.com/fish-shell/fish-shell/pull/10351 for future readers for the benefit of future readers
Looks good to me overall.

I think we don't even need to define `prompt_start` fields, can simply hardcode them at the callsite.
Also make sure to remove unused code.

Reply:
if the output is not a terminal, what will happen if it receives those escseq